cloud-init-configs/ubuntu/spin-up-jammy.sh

```diff
@@ -29,19 +40,23 @@ else
   wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img
 fi
 
-cat << EOF > user-data
-#cloud-config
-
-users:
-  - default
-  - name: brian
-    sudo: ["ALL-(ALL) NOPASSWD:ALL"]
-    shell: /bin/sh
-    ssh-authorized-keys:
-      - ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAINbliisKG/Rajx1joDff9Q6Kh9tLL+KD9Zqe4ZOsVttU b.r.reed@gmail.com
-EOF
+if [ -s my-seed.img ]; then
+  echo "my-seed.img has been supplied with data"
+else
+  echo "Supplying my-seed.img with data"
+  cloud-localds my-seed.img user-data meta-data
+fi
 
-cloud-localds my-seed.img user-data meta-data
+if [ -f ./ub-jammy-vm.qcow2 ]; then
+  echo "ub-jammy-vm.qcow2 already exists"
+else
+  echo "creating ub-jammy-vm.qcow2"
+  qemu-img create -F qcow2 -b jammy-server-cloudimg-amd64.img -f qcow2 ./ub-jammy-vm.qcow2 50G
+fi
 
-qemu-img create -F qcow2 -b jammy-server-cloudimg-amd64.img -f qcow2 ./ub-jammy-vm.qcow2 50G
-virt-install --name jammy-cloud --memory 4096 --vcpus 4 --os-variant ubuntujammy --disk ./ub-jammy-vm.qcow2 --disk ./my-seed.img --import
\ No newline at end of file
+if (virsh domifaddr jammy-cloud) then
```

Comment:
You could also do `virsh list | grep running | grep VMNAME`.

Reply:
changed -- 

Should I be using a variable for the vm name?